libcudacxx/include/cuda/std/__type_traits/conditional.h

```diff
@@ -53,8 +33,35 @@ struct _CCCL_TYPE_VISIBILITY_DEFAULT conditional<false, _If, _Then>
   typedef _Then type;
 };
 
-template <bool _Bp, class _If, class _Then>
-using conditional_t _CCCL_NODEBUG_ALIAS = typename conditional<_Bp, _If, _Then>::type;
+#if defined(_CCCL_COMPILER_MSVC)
+
+template <bool _Cond, class _If, class _Else>
+using conditional_t _CCCL_NODEBUG_ALIAS = typename conditional<_Cond, _If, _Else>::type;
+
+#else // ^^^ _CCCL_COMPILER_MSVC ^^^ / vvv !_CCCL_COMPILER_MSVC vvv
+
+// Optimized implementation of `conditional_t` instantiating only two classes
+template <bool>
+struct __conditional_impl;
+
+template <>
+struct __conditional_impl<true>
+{
+  template <class _If, class _Else>
+  using type _CCCL_NODEBUG_ALIAS = _If;
+};
+
+template <>
+struct __conditional_impl<false>
+{
+  template <class _If, class _Else>
+  using type _CCCL_NODEBUG_ALIAS = _Else;
```

Comment:
it doesn't much matter in this case, but our metaprogramming utilities like to call this nested alias template `__call`. then `conditional_t` could be an alias for `__type_call2<__conditional_impl<_Cond>, _If, _Else>`.